pkg/cmd/compile/options.go

```diff
@@ -126,10 +126,18 @@ func (o *CompileOptions) PreSet(preCheck func(cur string) bool) {
 	}
 
 	if len(o.Settings) == 0 {
-		o.Settings = []string{filepath.Join(projectstack.CiTestDir, projectstack.SettingsFile), projectstack.KclFile}
+		o.Settings = []string{projectstack.KclFile}
+		if _, err := os.Stat(filepath.Join(curDir, projectstack.CiTestDir, projectstack.SettingsFile)); err == nil {
+			o.Settings = append(o.Settings, filepath.Join(projectstack.CiTestDir, projectstack.SettingsFile))
+		}
```

Comment:
If err of os.Stat is not nil and not the NotExist error, should log it. Use level Warn is a good choice.